lib/Sema/CSBindings.cpp

```diff
@@ -224,23 +224,24 @@ void ConstraintSystem::PotentialBindings::addPotentialBinding(
 
 bool ConstraintSystem::PotentialBindings::isViable(
     PotentialBinding &binding) const {
-  // Prevent against checking against the same bound generic type
+  // Prevent against checking against the same opened nominal type
   // over and over again. Doing so means redundant work in the best
   // case. In the worst case, we'll produce lots of duplicate solutions
   // for this constraint system, which is problematic for overload
   // resolution.
   auto type = binding.BindingType;
   if (type->hasTypeVariable()) {
-    auto *BGT = type->getAs<BoundGenericType>();
-    if (!BGT)
+    auto *NT = type->getAs<NominalOrBoundGenericNominalType>();
```

Comment:
A cleaner way to write this would be type->getAnyNominal()

Reply:
Thanks!